### PR TITLE
Api800/uplink set

### PR DESCRIPTION
**Pull request by Sowmyahsg**
Description
Support for Uplink Sets until API800 , 500 and 600

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [x] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: ov/uplink_set.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+type PortConfigInfos struct {
+	DesiredSpeed string   `json:"desiredSpeed,omitempty"` //"desiredSpeed":"Auto",
+	Location     Location `json:"location"`               //"location":"{...},
+}
+
+type UplinkSet struct {
+	Name                           string            `json:"name,omitempty"`                           // "name": "Uplink77",
+	LogicalInterconnectURI         utils.Nstring     `json:"logicalInterconnectUri,omitempty"`         // "logicalInterconnectUri": "/rest/logical-interconnects/7769cae0-b680-435b-9b87-9b864c81657f",
+	NetworkURIs                    []utils.Nstring   `json:"networkUris,omitempty"`                    // "networkUris": "/rest/uplink-sets/e2f0031b-52bd-4223-9ac1-d91cb519d548",
```
Comment (madhav-bharadwaj) [inline ov/uplink_set.go]: Can have network specific uri in example.

File: ov/uplink_set.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+	)
+	upSets, err := c.GetUplinkSets(fmt.Sprintf("name matches '%s'", name), "name:asc")
+	if upSets.Total > 0 {
+		return upSets.Members[0], err
+	} else {
+		return upSet, err
+	}
+}
+
+func (c *OVClient) GetUplinkSets(filter string, sort string) (UplinkSetList, error) {
```
Comment (madhav-bharadwaj) [inline ov/uplink_set.go]: Maybe we can add support for start and count also as it is available as part of api800

File: ov/uplink_set.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+	return uplinkSets, nil
+}
+
+func (c *OVClient) GetUplinkSetById(id string) ([]string, error) {
+	var (
+		uri         = "/rest/uplink-sets/"
+		uplinkSetId = new([]string)
+	)
+	uri = uri + id
+	fmt.Println(uri)
```
Comment (madhav-bharadwaj) [inline ov/uplink_set.go]: Do we need to print the uri here?

File: ov/uplink_set.go [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+		uri = "/rest/uplink-sets"
+		t   *Task
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t = t.NewProfileTask(c)
+	t.ResetTask()
+	log.Infof("REST : %s \n %+v\n", uri, upSet)
```
Comment (madhav-bharadwaj) [inline ov/uplink_set.go]: Can change it back to Debug as we dont want it printed unless the user requires it.

File: ov/uplink_set.go [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+	//log.Infof("%s",json.Marshal(*upSet))
```
Comment (madhav-bharadwaj) [inline ov/uplink_set.go]: Can remove this comment if not needed.

File: ov/uplink_set.go [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+	uri = uri + id
+	fmt.Println(uri)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return *uplinkSetId, err
+	}
+	log.Infof("GetUplinkSetId %s", data)
```
Comment (madhav-bharadwaj) [inline ov/uplink_set.go]: Do we need infof here?

File: ov/uplink_set.go [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+	//log.Infof("%s",json.Marshal(*upSet))
+	log.Debugf("task -> %+v", t)
+	data, err := c.RestAPICall(rest.POST, uri, upSet)
+	if err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error submitting new Uplink Set request: %s", err)
+		return err
+	}
+
+	log.Infof("Response New Uplink Set %s", data)
```
Comment (madhav-bharadwaj) [inline ov/uplink_set.go]: Same as above. Do we need infof?

File: examples/uplink_set.go [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+	}
+
+	networkUris := new([]utils.Nstring)
+	*networkUris = append(*networkUris, utils.NewNstring("/rest/ethernet-networks/cbde97d0-c8f1-4aba-aa86-2b4e5d080401"))
+	//      Using make as the [] is going as null
+	fcNetworkUris := make([]utils.Nstring, 0)
+	fcoeNetworkUris := make([]utils.Nstring, 0)
+	portConfigInfos := make([]ov.PortConfigInfos, 0)
+
+	uplinkSet := ov.UplinkSet{Name: "upset77", LogicalInterconnectURI: utils.NewNstring("/rest/logical-interconnects/d4468f89-4442-4324-9c01-624c7382db2d"), NetworkURIs: *networkUris, FcNetworkURIs: fcNetworkUris, FcoeNetworkURIs: fcoeNetworkUris, PortConfigInfos: portConfigInfos, ConnectionMode: "Auto", NetworkType: "Ethernet", EthernetNetworkType: "Tagged", Type: "uplink-setV4", ManualLoginRedistributionState: "NotSupported"}
```
Comment (jyothis-gireesan) [inline examples/uplink_set.go]: We can make this struct parameters so as to increase the readability for the user.

**Pull request merged**